### PR TITLE
feat: add message about unconfigured feature flags when user runs `cdk synth`

### PR DESCRIPTION
**Pull request by vivian12300**
#### Notify users about the number of unconfigured feature flags in their application and the `cdk flags` command when they run `cdk synth`.
---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license

### CODE REVIEW

File: packages/aws-cdk/lib/cli/cli.ts [hunk 1]
```diff
@@ -546,6 +547,28 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
   }
 }
 
+export async function displayFlagsMessage(ioHost: CliIoHost, toolkitStackName: string, cloudExecutable: CloudExecutable,
```
Comment (otaviomacedo) [inline packages/aws-cdk/lib/cli/cli.ts]: You can move this logic inside `CdkToolkit`, where you already have access to the `Toolkit` instance.

File: packages/aws-cdk/lib/cli/cli.ts [hunk 2]
```diff
@@ -546,6 +547,28 @@ export async function exec(args: string[], synthesizer?: Synthesizer): Promise<n
...
+  ioHelper: IoHelper): Promise<void> {
+  const tk = new Toolkit({
+    ioHost,
+    toolkitStackName,
+    unstableFeatures: ['flags'],
+  });
+  let flagData = await tk.flags(cloudExecutable);
+
+  const OBSOLETE_FLAGS = [
```
Comment (otaviomacedo) [inline packages/aws-cdk/lib/cli/cli.ts]: You had defined this constant somewhere else, also, hadn't you? See if you can deduplicate it by exporting it from some common module.

File: packages/aws-cdk/test/cli/display-flags-message.test.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    } as any;
+
+    jest.spyOn(Toolkit.prototype, 'flags').mockImplementation(mockToolkit.flags);
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  test('displays message with correct count of unconfigured flags, filtering out obsolete flags', async () => {
```
Comment (otaviomacedo) [inline packages/aws-cdk/test/cli/display-flags-message.test.ts:28]: It's also important to test the negative case, where there are no unconfigured flags.

**Review by otaviomacedo — CHANGES_REQUESTED** [review]

File: packages/aws-cdk/lib/cli/cdk-toolkit.ts [hunk 4]
```diff
@@ -2108,6 +2111,22 @@ async function askUserConfirmation(
...
   });
 }
+export async function displayFlagsMessage(ioHost: CliIoHost, toolkitStackName: string, cloudExecutable: CloudExecutable,
+  ioHelper: IoHelper): Promise<void> {
+  const tk = new Toolkit({
+    ioHost,
+    toolkitStackName,
+    unstableFeatures: ['flags'],
+  });
+  let flagData = await tk.flags(cloudExecutable);
```
Comment (otaviomacedo) [inline packages/aws-cdk/lib/cli/cdk-toolkit.ts]:
```suggestion
  let flagData = await this.toolkit.flags(cloudExecutable);
```

**Review by otaviomacedo — CHANGES_REQUESTED** [review]

File: packages/aws-cdk/lib/cli/cdk-toolkit.ts [hunk 5]
```diff
@@ -2108,6 +2111,17 @@ async function askUserConfirmation(
     }
   });
 }
+export async function displayFlagsMessage(toolkit: InternalToolkit, cloudExecutable: CloudExecutable,
+  ioHelper: IoHelper): Promise<void> {
+  let flagData = await toolkit.flags(cloudExecutable);
+
+  flagData = flagData.filter(flag => !OBSOLETE_FLAGS.includes(flag.name));
+
+  const numUnconfigured = flagData.filter(flag => flag.userValue === undefined).length;
```
Comment (otaviomacedo) [inline packages/aws-cdk/lib/cli/cdk-toolkit.ts]: Up to you, but I prefer to chain the calls like this:


```suggestion
  const numUnconfigured = await toolkit.flags(cloudExecutable)
                              .filter(flag => !OBSOLETE_FLAGS.includes(flag.name))
                              .filter(flag => flag.userValue === undefined).length;
```

**Review by otaviomacedo — APPROVED** [review]

**Review by iankhou — APPROVED** [review]
